### PR TITLE
chore: replace make with mage

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
Makefile is getting more complicated and mixing shell/Go scripts. [Mage](https://magefile.org/) is a make/rake-like build tool using Go. It allows us to rewrite shell scripts in Go, which is more readable for Go developers.

- https://github.com/aquasecurity/trivy/blob/f14bed453246a880854ff2ee8cdd1a483e4ebc4c/integration/scripts/download-images.sh
- https://github.com/aquasecurity/trivy/blob/f14bed453246a880854ff2ee8cdd1a483e4ebc4c/integration/scripts/download-vm-images.sh
- https://github.com/aquasecurity/trivy/blob/f14bed453246a880854ff2ee8cdd1a483e4ebc4c/ci/deploy-rpm.sh
- https://github.com/aquasecurity/trivy/blob/f14bed453246a880854ff2ee8cdd1a483e4ebc4c/ci/deploy-deb.sh
- etc.

It also helps https://github.com/aquasecurity/trivy/issues/3874.

```
$ mage -l
Targets:
  build                          builds Trivy
  clean                          cleans up the fixtures
  docs:generate                  generates CLI references
  docs:serve                     launches MkDocs development server to preview the documentation page
  easyjson                       generates JSON marshaler/unmarshaler for TinyGo/WebAssembly as TinyGo doesn't support encoding/json.
  fmt                            formats Go code and proto files
  install                        installs Trivy
  lint                           runs linters
  mock                           generates mocks
  protoc                         parses PROTO_FILES and generates the Go code for client/server mode
  test:fixtureContainerImages    downloads and extracts required images
  test:fixtureVMImages           downloads and extracts required VM images
  test:generateExampleModules    compiles example Wasm modules for integration tests
  test:generateModules           compiles WASM modules for unit tests
  test:integration               runs integration tests
  test:module                    runs Wasm integration tests
  test:unit                      runs unit tests
  test:vm                        runs VM integration tests
  tidy                           makes sure go.mod matches the source code in the module
  tool:aqua                      installs aqua if not installed
  tool:crane                     installs crane
  tool:easyJSON                  installs easyjson
  tool:golangciLint              installs golangci-lint
  tool:goyacc                    installs goyacc
  tool:labeler                   installs labeler
  tool:mockery                   installs mockery
  tool:wire                      installs wire if not installed
  wire                           generates the wire_gen.go file for each package
  yacc                           generates parser
```

## Related issues
- https://github.com/aquasecurity/trivy/issues/3874

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: magefiles/magefile.go [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+		}
+	}
+	return nil
+}
+
+// Unit runs unit tests
+func (t Test) Unit() error {
+	mg.Deps(t.GenerateModules)
+	return sh.RunWith(ENV, "go", "test", "-v", "-short", "-coverprofile=coverage.txt", "-covermode=atomic", "./...")
+}
```
Comment (DmitriyLewen) [inline magefiles/magefile.go:243]: can we add logs, like for make file?
```
➜  mage test:unit
Error: running "go test -v -short -coverprofile=coverage.txt -covermode=atomic ./..." failed with exit code 1

```

Comment (DmitriyLewen) [reply magefiles/magefile.go:243]: Same for other tests

Comment (knqyf263) [reply magefiles/magefile.go]: `mage -v` shows stdout, but I forced it for tests.
https://github.com/aquasecurity/trivy/pull/3932/commits/f325cc61c234172cd96036c1989d383a2ddd2a1c

**Review by DmitriyLewen — APPROVED** [review]

**Review by AnaisUrlichs — COMMENTED** [review]
looks good

**Pull request merged**